zerver/lib/addressee.py

```diff
@@ -170,7 +170,7 @@ def for_stream_name(stream_name: str, topic: str) -> 'Addressee':
         )
 
     @staticmethod
-    def for_stream_id(stream_id: int, topic: str) -> 'Addressee':
+    def for_stream_id(stream_id: int, topic: Optional[str]) -> 'Addressee':
```

Comment:
I'm not sure this is the right fix.  I think the clear intent from the code is that you should not be able to send `None` as the topic -- it's a code error that we're somewhat awkardly trying to handle in validate_topic..

Reply:
I've removed the dynamic check and changed this back to `str` in my local update.